__Euclidean distance will find the simularity between point/list of points by calculating the distance between then in euclidean space. Below is the mathematical notation of the formula-__

<img src="image/Euclidean Distance.png" style="width:400px;height:150px">

### Import Library and input file

In [19]:
import pandas as pd
import numpy as np
import math

In [10]:
df = pd.read_csv(os.getcwd()+'\input_file\iris.csv')
print(df['class'].unique())
df.head()

['Iris-setosa' 'Iris-versicolor' 'Iris-virginica']


sepal_length  sepal_width  petal_length  petal_width        class
0           5.1          3.5           1.4          0.2  Iris-setosa
1           4.9          3.0           1.4          0.2  Iris-setosa
2           4.7          3.2           1.3          0.2  Iris-setosa
3           4.6          3.1           1.5          0.2  Iris-setosa
4           5.0          3.6           1.4          0.2  Iris-setosa

__We'll take random sample from each of the iris class and compare which other classes are closer to sentosa__

In [15]:
setosa     = df.loc[df['class']=='Iris-setosa'    ,['sepal_length','sepal_width','petal_length','petal_width']].sample(1).values[0]
versicolor = df.loc[df['class']=='Iris-versicolor',['sepal_length','sepal_width','petal_length','petal_width']].sample(1).values[0]
virginica  = df.loc[df['class']=='Iris-virginica' ,['sepal_length','sepal_width','petal_length','petal_width']].sample(1).values[0]

In [17]:
print(setosa)
print(versicolor)
print(virginica)

[5.7 3.8 1.7 0.3]
[6.  3.4 4.5 1.6]
[7.7 3.  6.1 2.3]


In [20]:
def euclidean_distance (C1, C2):
    
    return math.sqrt( (C1[0] - C2[0])**2 + 
                      (C1[1] - C2[1])**2 + 
                      (C1[2] - C2[2])**2
                    )

In [23]:
dist_setosa_vs_versicolor = euclidean_distance(setosa,versicolor)
dist_setosa_vs_virginica  = euclidean_distance(setosa,virginica)

In [24]:
print('setosa_vs_versicolor     - {}'.format(dist_setosa_2_versicolor))
print('dist_setosa_vs_virginica - {}'.format(dist_setosa_2_virginica))

setosa_vs_versicolor     - 2.844292530665578
dist_setosa_vs_virginica - 4.898979485566356


__From the distance we can see sentosa & versicolor are very similar (Dist=2.84) in terms of their shape compare to verginica (Dist=4.89). Find the similarity in the below image. *Lower the distance value closer/similar the variables are.*__
<img src="image/iris.png" style="width:600px;height:200px">